crates/typst-pdf/src/lib.rs

```diff
@@ -142,6 +154,29 @@ pub enum PdfStandard {
     /// PDF/A-2b.
     #[serde(rename = "a-2b")]
     A_2b,
+    /// PDF/A-3b.
+    #[serde(rename = "a-3b")]
+    A_3b,
+}
+
+/// Conformance levels of PDF/A
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[allow(non_camel_case_types)]
+#[non_exhaustive]
+pub enum PdfAConformanceLevel {
```

Comment:
This should not be public. We intentionally do not provide such an enum to keep the API flexible. Internally, we can turn `&[PdfStandard]` into whatever is most convenient at the time, rejecting bad combinations.

I think as is the enum is a bit strange because it drops the "b".

As various checks are required by different standards (e.g. all your `is_some()`) checks, another design might be to not store level enums, but rather a bunch of booleans specifying which checks should be performed. 

If we like this bool-based design, maybe for now we could realize it with just `pdfa: bool, embedded_files: bool` and later break out the `pdfa: bool` into separate ones. And for the XMP metadata we could just store a field `(u8, &'static str)` (e.g. `(2, "B")`).

Reply:
I removed the enum, but in my opinion using two booleans makes the code harder to understand in some places. If we need to differentiate between 2B and 3B, we now use the fact that we disallow file embeddings for 2B.

For example: when we write the `pdfa_part` in the catalog, we now use the `embedded_files` boolean to differentiate between 2B and 3B.

Reply:
For the pdfa_part in the catalog, I wanted to have the `(u8, &'static str)` as `ctx.options.standards.part`. Essentially everything that is semantically separate becomes its own thing in `PdfStandards`.